Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles.cs

```diff
@@ -1,220 +1,214 @@
-
-                return new Result(result);
+                fileTransferLog.Warning("'{0}' is not a valid task execution ID, will default to random Guid", info.TaskExecutionID);
+                executionId = Guid.NewGuid();
             }
-        }
 
-        /// <summary>
-        /// Fetches log lines in case of exception.
-        /// </summary>
-        /// <param name="buffer"></param>
-        /// <returns></returns>
-        private static string GetLogLines(IEnumerable<Tuple<DateTimeOffset, string>> buffer)
-        {
-            try
+            _batchContext = new BatchContext
             {
-                return string.Join("\n", buffer.Select(x => x.Item1 == DateTimeOffset.MinValue ? "..." : $"{x.Item1:HH:mm:ssZ}: {x.Item2}"));
+                Info = info,
+                Options = options,
+                InstanceId = executionId,
+                ServiceId = info.TransferName,
+                SourceFiles = new List<FileItem>(),
+                DestinationFiles = new List<SftpFile>(),
+                RoutineUri = info.ProcessUri,
+                BatchTransferStartTime = DateTime.Now,
+                Source = source,
+                Destination = destination,
+                Connection = connection
+            };
 
-            }
-            catch (Exception e)
-            {
-                return $"Error while creating log: \n{e.ToString()}";
-            }
-        }        
+            var fileTransporter = new FileTransporter(logger, _batchContext, executionId);
+            var result = fileTransporter.Run(cancellationToken);
 
-        private static SFTPLogger InitializeSFTPLogger(ILogger notificationLogger)
-        {
-            var logger = new SFTPLogger(notificationLogger);
-            return logger;
+            if (options.ThrowErrorOnFail && !result.Success)
+                throw new Exception($"SFTP transfer failed: {result.UserResultMessage}. " +
+                                    $"Latest operations: \n{GetLogLines(transferSink.GetBufferedLogMessages())}");
+
+            if (options.OperationLog)
+                result.OperationsLog = GetLogDictionary(transferSink.GetBufferedLogMessages());
+
+            return new Result(result);
         }
+    }
+
+    /// <summary>
+    /// Fetches log lines in case of exception.
+    /// </summary>
+    /// <param name="buffer"></param>
+    /// <returns></returns>
+    private static string GetLogLines(IEnumerable<Tuple<DateTimeOffset, string>> buffer)
+    {
+        try
+        {
+            return string.Join("\n", buffer.Select(x => x.Item1 == DateTimeOffset.MinValue ? "..." : $"{x.Item1:HH:mm:ssZ}: {x.Item2}"));
 
-        private static IDictionary<string, string> GetLogDictionary(IList<Tuple<DateTimeOffset, string>> entries)
+        }
+        catch (Exception e)
         {
-            const string dateFormat = "yyyy-MM-dd HH:mm:ss.f0Z";
+            return $"Error while creating log: \n{e.ToString()}";
```

Comment:
When you are using string interpretation, if the value is not a string, compiler will automatically call the objects ToString-method, so you can change `{e.ToString()}` to `{e}`.

Reply:
Fixed.